configure.ac

```diff
@@ -2082,6 +2082,35 @@ case $CC in
         fi
 esac
 
+dnl Historically, some of our code assumed that signed integer overflow
+dnl is defined behaviour via twos-complement.
+dnl Set STRICT_OVERFLOW_CFLAGS and NO_STRICT_OVERFLOW_CFLAGS depending on compiler support.
+dnl Pass the latter to modules that depend on such behaviour.
+_SAVE_VAR([CFLAGS])
+CFLAGS="-fstrict-overflow -fno-strict-overflow"
+AC_CACHE_CHECK([if $CC supports -fstrict-overflow and -fno-strict-overflow],
+               [ac_cv_cc_supports_fstrict_overflow],
+  AC_COMPILE_IFELSE(
+    [
+      AC_LANG_PROGRAM([[]], [[]])
+    ],[
+      STRICT_OVERFLOW_CFLAGS="-fstrict-overflow"
+      NO_STRICT_OVERFLOW_CFLAGS="-fno-strict-overflow"
+    ],[
+      STRICT_OVERFLOW_CFLAGS=""
+      NO_STRICT_OVERFLOW_CFLAGS=""
+    ])
+)
+_RESTORE_VAR([CFLAGS])
+
+AC_MSG_CHECKING([for --with-strict-overflow])
+AC_ARG_WITH([strict-overflow],
+  AS_HELP_STRING(
+    [--with-strict-overflow],
+    [choose between -fstrict-overflow or -fno-strict-overflow (default is no)]),[],
+    [with_strict_overflow=no])
```

Comment:
Sorry to pick on this further. I would like two more changes before merging:

1. The help message is too vague. I'd like something a la: _if 'yes', add -fstrict-overflow to CFLAGS, else add -fno-strict-overflow to CFLAGS (default is no)_.
2. Please add a `AC_MSG_WARN` if `ac_cv_cc_supports_fstrict_overflow=no` and `--with-strict-overflow=yes` (_"--with-strict-overflow=yes specified, but your compiler does not support the -fstrict-overflow flag"_

Reply:
Makes sense. Don't worry about picking on this, I'm just glad you care enough to make the effort!

Reply:
Thanks, just ping me when you've made the changes, and I'll get this merged.